File: app/assets/javascripts/OperationDefinition.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const OperationDefinitionSchema = DataElementSchema({
  url: Any,
  version: Any,
  name: Any,
  title: Any,
  status: Any,
  kind: Any,
  experimental: Any,
  date: Any,
  publisher: Any,
  contact: Any,
  description: Any,
  useContext: Any,
  jurisdiction: Any,
  purpose: Any,
  affectsState: Any,
  code: Any,
  comment: Any,
  base: Any,
  resource: Any,
  system: Any,
  type: Any,
  instance: Any,
  inputProfile: Any,
  outputProfile: Any,
  parameter: Any,
  overload: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'OperationDefinition' },

});

module.exports.OperationDefinitionSchema = OperationDefinitionSchema;
module.exports.OperationDefinition = mongoose.model('OperationDefinition', OperationDefinitionSchema);

File: app/models/qdm/operation_definition.rb
module QDM
  # app/models/qdm/operation_definition.rb
  class OperationDefinition < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :url
    field :version
    field :name
    field :title
    field :status
    field :kind
    field :experimental
    field :date
    field :publisher
    field :contact
    field :description
    field :useContext
    field :jurisdiction
    field :purpose
    field :affectsState
    field :code
    field :comment
    field :base
    field :resource
    field :system
    field :type
    field :instance
    field :inputProfile
    field :outputProfile
    field :parameter
    field :overload
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
